core/gui/src/app/workspace/component/menu/menu.component.ts

```diff
@@ -248,6 +249,21 @@ export class MenuComponent implements OnInit {
     this.executeWorkflowService.takeGlobalCheckpoint();
   }
 
+  /**
+   * get the html to export all results.
+   */
+  public onClickExportAllResults(): void {
+    this.reportPrintService
+      .getWorkflowSnapshot()
+      .pipe(
+        tap((snapshot: string) => this.reportPrintService.downloadResultsAsHtml(snapshot)),
```

Comment:
put this downloadResultAsHtml logic to `subscribe`, for pipe, only keep `untilDestroy`